File: src/main/java/ru/serdtsev/homemoney/moneyoper/model/MoneyOperItemDto.java
package ru.serdtsev.homemoney.moneyoper.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Data @AllArgsConstructor
public class MoneyOperItemDto {
  private UUID id;
  private UUID balanceId;
  private String balanceName;
  private BigDecimal value;
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
  private LocalDate performedAt;
  private int index;
}

File: src/main/kotlin/ru/serdtsev/homemoney/moneyoper/model/MoneyOperItemDto.kt
package ru.serdtsev.homemoney.moneyoper.model

import java.math.BigDecimal
import java.time.LocalDate
import java.util.*

data class MoneyOperItemDto (
        val id: UUID,
        val balanceId: UUID,
        val balanceName: String?,
        val value: BigDecimal,
        val currencyCode: String,
        val performedAt: LocalDate?,
        val index: Int = 0
) {
    fun getCurrencySymbol() = Currency.getInstance(currencyCode).symbol
}
